cmake/strip.cmake

```diff
@@ -0,0 +1,18 @@
+# * Strips debug symbols from an executable target
+# * Copies debug symbols into an extra debug-symbol-file
+# * Adds reference to this debug symbol from to the executable
+# * Does not support Windows / Is not needed on Windows
+function(strip_symbols target)
+  if(NOT WIN32)
+    add_custom_command(
+            TARGET ${target} POST_BUILD
+      COMMAND ${CMAKE_OBJCOPY} --only-keep-debug "$<TARGET_FILE_NAME:${target}>"
+              "$<TARGET_FILE_NAME:${target}>.debug"
```

Comment:
This is not strong opinion, but maybe it makes sense to keep it just unstripped. The workflow in mind is I want to upload this binary to server and run gdb on it, I can take this instead of stripped one. But I do not have strong opinion on this.

Reply:
I also thought about this and came to the same conclusion as you in the other comment.

Furthermore we should definitely strip the symbols for the package we deploy to the gamelet
since this might unnecessarily increase deployment times. So either we copy both files for debugging or have a second executable with debug symbols included which can lead to other
problems when someone accidentally copies the wrong file.

Reply:
As long as the symbols are always readily available, I'm fine with stripping them.  If one doesn't care about having symbols remotely, there's no obligation to push them.  We should have proper symbol servers set up anyway.  Personally, I'll have my "deploy" script always push debug symbols remotely.